common/scala/src/main/scala/org/apache/openwhisk/core/entity/ExecManifest.scala

```diff
@@ -344,9 +373,38 @@ protected[core] object ExecManifest {
 
   protected[entity] implicit val imageNameSerdes: RootJsonFormat[ImageName] = jsonFormat4(ImageName.apply)
 
-  protected[entity] implicit val stemCellSerdes: RootJsonFormat[StemCell] = {
+  protected[entity] implicit val ttlSerdes: RootJsonFormat[FiniteDuration] = new RootJsonFormat[FiniteDuration] {
+    override def write(finiteDuration: FiniteDuration): JsValue = JsString(finiteDuration.toString)
+
+    override def read(value: JsValue): FiniteDuration = value match {
+      case JsString(s) =>
+        val duration = Duration(s)
+        FiniteDuration(duration.length, duration.unit)
+      case _ =>
+        deserializationError("time unit not supported. Only milliseconds, seconds, minutes, hours, days are supported")
+    }
+  }
+
+  protected[entity] implicit val reactivePrewarmingConfigSerdes: RootJsonFormat[ReactivePrewarmingConfig] = jsonFormat5(
+    ReactivePrewarmingConfig.apply)
+
+  protected[entity] implicit val stemCellSerdes = new RootJsonFormat[StemCell] {
     import org.apache.openwhisk.core.entity.size.serdes
-    jsonFormat2(StemCell.apply)
+    val defaultSerdes = jsonFormat3(StemCell.apply)
+    override def read(value: JsValue): StemCell = {
+      val fields = value.asJsObject.fields
+      val initialCount =
+        fields
+          .get("initialCount")
+          .orElse(fields.get("count"))
+          .map(_.convertTo[Int])
+          .get
+      val memory = fields.get("memory").map(_.convertTo[ByteSize]).get
```

Comment:
I would avoid using `get` on `Option`.

Since these configurations are required, I think we can do something simliar to this.
```suggestion
    override def read(value: JsValue): StemCell = {
      val fields = value.asJsObject.fields
      val initialCount: Option[Int] =
        fields
          .get("initialCount")
          .orElse(fields.get("count"))
          .map(_.convertTo[Int])

      val memory: Option[ByteSize] = fields.get("memory").map(_.convertTo[ByteSize])
      val config = fields.get("reactive").map(_.convertTo[ReactivePrewarmingConfig])
      
      (initialCount, memory) match {
        case (Some(c), Some(m)) => StemCell(c, m, config)
        case (Some(c), None)  => throw new IllegalArgumentException(s"blah blah: $c")
        case (None, Some(m))  => throw new IllegalArgumentException(s"blah blah: $m")
        case _ => throw new IllegalArgumentException(s"blah blah")
      }
```

Reply:
updated accordingly.